From f5962f17ffe1a0a85f58cf04f2775a166dd82bba Mon Sep 17 00:00:00 2001
From: Ruijun Ge <geruijun@huawei.com>
Date: Mon, 27 Dec 2021 19:43:25 +0800
Subject: [PATCH 211/263] [Huawei] PCI: fix the wrong class type for HiSilicon
 NP 5896 again

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188816

--------------------------------

Commit 46bbc56a4f6a ("PCI: Add quirk for HiSilicon PCI Network Processor
devices") reverts the modification of commit 08aa7bae839e ("PCI: fix the
wrong class type for HiSilicon NP 5896 again") by mistake. Let's add it
back.

The following patch set the class type as 'PCI_BASE_CLASS_NETWORK'. But
'PCI_BASE_CLASS_NETWORK' is actually the higher 8 bits of the class type
of a network device. We should set it as 'PCI_CLASS_NETWORK_ETHERNET'.
This patch fixes it.

Fixes: 46bbc56a4f6a ("PCI: Add quirk for HiSilicon PCI Network Processor devices")

Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Ruijun Ge <geruijun@huawei.com>
Reviewed-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
---
 drivers/pci/quirks.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index 8bbe1d44799d..04fbf0928277 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -5844,7 +5844,7 @@ DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_NVIDIA, 0x0ab8, nvidia_ion_ahci_fixup);
 
 static void pci_quirk_hisi_fixup_class(struct pci_dev *dev)
 {
-	dev->class = PCI_BASE_CLASS_NETWORK << 8;
+	dev->class = PCI_CLASS_NETWORK_ETHERNET << 8;
 	pci_info(dev, "force hisi class type to network\n");
 }
 
-- 
2.17.1

